src/content/docs/cheerpj3/11-guides/Networking.md

```diff
@@ -0,0 +1,77 @@
+
+### Why Tailscale?
+
+To support networking beyond same origin requests, it is necessary to host a proxy server. CheerpJ cannot become responsible for the user's traffic as every application is different and its networking traffic should be managed internally and according to their needs.
+
+When it comes to generalised networking scenarios, there are some browser-enforced limitations to enhance users' security. These limitations are bound to the browser not exposing access to lower-level protocols such as UDP and TCP. The browser only allows one to perform HTTP(S) requests ruled by CORS policies.
+
+A good example of these scenarios is when an application uses WebSockets, where there is a two-way interaction between two end-points in an event-driven manner. A WebSocket connection starts with a request from the browser via HTTP that when accepted, this connection is upgraded and does not conform to the HTTP protocol anymore. This way the user and the server (or any other end-point) can keep a connection and send bidirectional messages until this connection is closed from one of the end-points. Upgrading the WebSocket connection protocol and unwrapping packets before sending them to the destinations requires a proxy server.
+
+These limitations have been overcome by supporting networking via Tailscale, which allows one to use its VPN via its WebSocket proxy, meaning the perfect solution for the limitations described above.
+
+### Installing Tailscale
+
+The easiest way to enable generalized networking on your Java application without modifying the source code is to [install Tailscale](https://tailscale.com/kb/1017/install) on your local server. This way the client can connect to your private Tailscale network via an auth key, and then talk to your server via the VPN.
+
+### Connecting your application to a Tailscale network
+
+Connecting the Java application client to your Tailscale network is as simple as providing client authentication via `cheerpjInit()`:
+
+Example for pre-authenticated nodes:
+
+```js
+cheerpjInit({
+	tailscaleControlUrl: "https://my.url.com/",
+	tailscaleAuthKey: "AuthKeyStringGoesHere",
+});
+```
+
+What is happening here?
+
+- `tailscaleControlUrl` is a string URL of the Tailscale control plane which verifies the user's identity. Only pass this option if you are [self-hosting Tailscale](https://github.com/leaningtech/headscale)
+- `tailscaleAuthKey` is string with an auth key to register new nodes that are pre-authenticated. You can create an auth key [here](https://login.tailscale.com/admin/settings/keys).
+
+Example to prompt the user for manual login:
+
+```js
+cheerpjInit({
+	tailscaleControlUrl: "https://my.url.com/",
+	tailscaleLoginUrlCb: function (url) {
+		// your function code here to continue with login
```

Comment:
An actual example here that returns something would be good. Maybe see webvm for an example

Reply:
I think tailscaleLoginUrlCb does not return anything, but I added a few lines on how it would be to prompt the login to a different tab for now (based on the WebVM example). Will add something more complex as I play with it.

Reply:
My bad. Thanks, lgtm